src/beliefpropagation/beliefpropagation.jl

```diff
@@ -102,7 +103,43 @@ function belief_propagation_iteration(
       c += 0.5 * norm(denseblocks(LHS) - denseblocks(RHS))
     end
   end
-  return new_mts, c / (length(es))
+  return new_mts, c / (length(edges))
+end
+
+"""
+Do parallel updates between groups of edges of all message tensors for a given ITensornetwork and its partition into sub graphs
+"""
+function belief_propagation_iteration(
+  tn::ITensorNetwork,
+  mts::DataGraph,
+  edge_groups::Vector{Vector{E}};
```

Comment:
```suggestion
  edge_groups::Vector{Vector{<:AbstractEdge}};
```
In general I think we should use `AbstractEdge` instead of `NamedEdge` for dispatch where possible to make the code more generic.